controller/src/main/java/io/pravega/controller/store/PravegaTablesStoreHelper.java

```diff
@@ -244,6 +249,89 @@ public void invalidateCache(String table, String key) {
+                    updatedBytes = TagRecord.builder().tagName(k).build().toBytes();
+                } else {
+                    TagRecord record = TagRecord.fromBytes(array);
+                    TagRecord tep = record.toBuilder().removeStream(removeValue).build();
+                    updatedBytes = tep.toBytes();
+                }
+                return TableSegmentEntry.versioned(k.getBytes(StandardCharsets.UTF_8),
+                                                   updatedBytes,
+                                                   entry.getKey().getVersion().getSegmentVersion());
+
+            }).collect(Collectors.toList());
+            return segmentHelper.updateTableEntries(tableName, updatedList, authToken.get(), requestId)
+                                .whenComplete((v, ex) -> releaseEntries(updatedList));
+        });
+        return CompletableFuture.completedFuture(Collections.emptyList());
+    }
+
+    private CompletableFuture<List<TableSegmentKeyVersion>> appendValuesTable(String tableName, String appendValue, Set<String> tableKeys, long requestId) {
+        // Get unversioned keys.
+        final List<TableSegmentKey> keys = tableKeys.stream()
+                                                    .map(k -> TableSegmentKey.unversioned(k.getBytes(StandardCharsets.UTF_8)))
+                                                    .collect(Collectors.toList());
+        return segmentHelper.readTable(tableName, keys, authToken.get(), requestId) // read the latest version of keys.
+                            .whenComplete((v, ex) -> releaseKeys(keys))
+                            .thenCompose(currentEntries -> {
+
+                                List<TableSegmentEntry> updatedList = currentEntries.stream().map(entry -> {
+                                    String k = entry.getKey().getKey().toString(StandardCharsets.UTF_8);
+                                    byte[] array = getArray(entry.getValue());
+                                    byte[] updatedBytes;
+                                    if (array.length == 0) {
+                                        updatedBytes = TagRecord.builder().tagName(k).stream(appendValue).build().toBytes();
+                                    } else {
+                                        TagRecord record = TagRecord.fromBytes(array);
+                                        updatedBytes = record.toBuilder().stream(appendValue).build().toBytes();
+                                    }
+                                    return TableSegmentEntry.versioned(k.getBytes(StandardCharsets.UTF_8),
+                                                                       updatedBytes,
+                                                                       entry.getKey().getVersion().getSegmentVersion());
+
+                                }).collect(Collectors.toList());
+                                return segmentHelper.updateTableEntries(tableName, updatedList, authToken.get(), requestId)
```

Comment:
i think the reason you have put this in here is because we dont have updateTableEntries as a store helper method. 
I suggest you add that helper method which can take a batch of keys and values and update them. 
and then you can write your higher level business logic to use it.
also this updateEntries should be a generalized form of udateEntry method where we do it for more than one key value pair. maybe we can restrict its implementation to expect a single value type.

Reply:
done.